### PR TITLE
Allow attachment of worker to Dead-letter queue

### PR DESCRIPTION
**Pull request by whambulance**
References Issue #188 

Changes aimed at allowing a worker to be assigned as a handler to the dead letter queue. Also publishing the dlq ARN to the yaml file - similar to how the main queue ARN 

Reasons:
* It was difficult to add a worker to a DLQ compared to how easy it was to add one to the main queue
* It would have required manually inserting the ARN into the YAML after having created the queue
* Allowing additional manipulation of the DLQ due to exposure of it's ARN in the YAML file

### CODE REVIEW

**Comment by mnapoli** [comment]
Hi, I'm curious why you would want to have a worker on the DLQ?

**Comment by whambulance** [comment]
Hi @mnapoli - In my specific scenario, I'm using SQS to handle making api calls to a third party provider. If their API is unavailable for some reason, the messages will get pushed into the DLQ, and I need to be able to retry these failed messages at a second provider. This means changing up the content of the message.

[Here's another example from the AWS blog](https://aws.amazon.com/blogs/compute/using-amazon-sqs-dead-letter-queues-to-replay-messages/), written by one of their Senior solution architects.

Essentially, there exists use cases where simply pushing the failed messages back into the main queue isn't viable, and additional processing is required.

**Comment by mnapoli** [comment]
Thanks for the explanation.

I think that falls outside of the scope of the construct, as it is right now. I don't see that feature being merged at the moment.

Maybe #128 is an alternative that has more chance of happening. Would that work for you?

**Comment by whambulance** [comment]
As far as the base framework goes, exposing the URL allows you to send messages into the queue, but in order to consume messages from it I believe the ARN needs to be exposed?

Sadly I don't think this would be a viable solution - unless it's modified to expose the DLQ ARN too, similar to what I've done in this PR.

**Comment by whambulance** [comment]
@mnapoli Updated to remove the dlqWorker construct, now only exposes dead letter queue URL & ARN

File: src/constructs/aws/Queue.ts [hunk 1]
```diff
@@ -251,6 +252,10 @@ export class Queue extends AwsConstruct {
             description: `URL of the "${id}" SQS queue.`,
             value: this.queue.queueUrl,
         });
+        this.dlqArnOutput = new CfnOutput(this, "DlqArn", {
+            description: `ARN of the "${id}" SQS dead letter queue.`,
+            value: this.dlq.queueArn,
+        });
```
Comment (mnapoli) [inline src/constructs/aws/Queue.ts:258]: This CloudFormation output is not needed as far as I know.

File: src/constructs/aws/Queue.ts [hunk 2]
```diff
@@ -119,6 +119,7 @@ export class Queue extends AwsConstruct {
     private readonly dlq: CdkQueue;
     private readonly queueArnOutput: CfnOutput;
     private readonly queueUrlOutput: CfnOutput;
+    private readonly dlqArnOutput: CfnOutput;
```
Comment (mnapoli) [inline src/constructs/aws/Queue.ts:122]: This variable becomes useless.

**Review by mnapoli — COMMENTED** [review]
Thanks for the update, that looks good! I have added a few comments inline.

Could you also rebase on master? The tests are now passing on master, and this should allow us to test this PR.

File: src/constructs/aws/Queue.ts (hunk 1 above)
Comment (mnapoli) [reply src/constructs/aws/Queue.ts:258]: Right, but the `${construct:jobs.dlqArn}` is transformed into a `!Ref` (cloudformation reference), it's not using the CF output.

The variables are exposed in the `variables()` method (not via CF outputs -> these are used for the CLI output).

Comment (whambulance) [reply src/constructs/aws/Queue.ts]: Ah - yes my bad, I believed it was the other way around. I've pushed a change that removes the initializer and the dlqArnOutput prop

**Review by fredericbarthelet — APPROVED** [review]

**Comment by fredericbarthelet** [comment]
LGTM :) Merging and issuing release now ;) thanks @whambulance for the contribution !

**Pull request merged**